.github/workflows/image.yaml

```diff
@@ -0,0 +1,79 @@
+name: release
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+jobs:
+  hnc-manager:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Set Release Tag
+        run: |
+          TAG="${GITHUB_REF#refs/*/v}"
```

Comment:
What does this give you if you're running on `master`?

Reply:
It gives it as `refs/heads/master`  whereas when it's pushed to `v1`, it gives it as `1`. 

https://github.com/RealHarshThakur/hierarchical-namespaces/runs/2753272153?check_suite_focus=true#step:8:11
https://github.com/RealHarshThakur/hierarchical-namespaces/runs/2753141639?check_suite_focus=true

Reply:
We could also not push image for master branch. Only when a release is made and a branch is created, we create an image for it. We can have a separate image tag for Github Actions, just hope it doesn't confuse the end users.  WDYT?

Reply:
This applies to the release branches as well. It's _only_ Git tags that only get built once.

I'd prefer if we could append some unique prefix onto each build that comes from a branch (and not a tag). Failing that, can we say `-latest` on it so people know not to rely on it being constant?

Reply:
Nit: I suggest doing a `+{date}` to the tag as these are automatic builds. A 'real' release should likely only be retagging an intermediate build into the proper version. At least that would be my suggestion, especially since our release include more than just an image build. It also includes docs, kustomization files, and soon to be helm charts.

Reply:
I added the date in the image tag but I removed the builds being triggered for master branch . Is that okay?

Reply:
@RealHarshThakur I'm not seeing the date in the tab above, am I missing someething?

Reply:
@rjbez17 added it [here ](https://github.com/kubernetes-sigs/hierarchical-namespaces/pull/45/commits/2f61d91f2c60221359d8e5e195734bbb235889cc#diff-b4df0a4f0d80f73138c476afbd7aefdac9df339642ddfba323d27c8cbabb92e2R50)

Reply:
I _think_ that sets the _Docker_ tag so that it includes the date, but `RELEASE_TAG` (which I'm guessing is the _Github_ release tag) still includes only the branch name, correct? E.g., I'd prefer for this to be something like:

```
GITREF="${GITHUB_REF#refs/*/v}"
echo "RELEASE_TAG=${GITREF}-${{ steps.date.outputs.HNC_BUILD_TIMESTAMP }}" >> $GITHUB_ENV
```

In other words, line up the `RELEASE_TAG` and the Docker tags as closely as possible.

@RealHarshThakur if I've misinterpreted please feel free to remove the hold. We can always clean it up later anyway.